File: app/src/main/kotlin/org/book2dictionary/Logger.java
package org.book2dictionary;

import android.util.Log;

import java.io.IOException;

public class Logger {

    private static final boolean debuggable = true;
    public static final String TAG = "book2words";

    public static void debug(String message) {
        debug(null, message);
    }

    public static void debug(String tag, String message) {
        if (debuggable) {
            if (tag != null) {
                message = String.format("%s - %s", tag, message);
            }
            Log.d(TAG, message);
        }
    }

    public static void info(String message) {
        Log.i(TAG, message);
    }

    public static void error(Throwable e) {
        error("", e);
    }

    public static void error(String message, Throwable e) {
        Log.e(TAG, message, e);
    }
}

File: app/src/main/kotlin/org/book2words/core/Logger.kt
package org.book2words.core

import android.util.Log

class Logger {
    companion object {

        private val debuggable = true
        val TAG: String = "book2words"

        fun debug(message: String, tag: String? = "") {
            if (debuggable) {
                Log.d(TAG, "$tag $message")
            }
        }

        fun info(message: String) {
            Log.i(TAG, message)
        }

        fun error(message: String) {
            Log.e(TAG, message, null)
        }

        fun error(e: Throwable) {
            error("", e)
        }

        fun error(message: String, e: Throwable) {
            Log.e(TAG, message, e)
        }
    }
}